Improve LSP0 and LSP14 and LSP17 Specification (#257)
lukso-network/LIPs: LSPs/LSP-17-ContractExtension.md

```json
{

>   "name": "LSP17Extension:<bytes4>",

Reviewer: Since we don't know how to represent the either `(address, bool)` or `address` for the value type + value content (awaiting reply from @frozeman) I would suggest we put both schemas and describe that it is possible to use either one or the other.

So we could keep the old schema here, and put the.second new one (this one with the tuple) under the following paragraph.

<img width="633" alt="image" src="https://github.com/lukso-network/LIPs/assets/31145285/6b226172-0b0b-4a8f-9ece-7699cdd28647">

But re-word it as:

_"A boolean at the end COULD be used to specify whether the value sent to the extendable contract should be sent to the extension or stay in the extendable contract. If this behaviour is desired, the implementation CAN use the following alternative ERC725Y JSON Schema:"_

  "key": "0xcee78b4094da860110960000<bytes4>",
  "keyType": "Mapping",

>   "valueType": "(address, bytes1)",

Reviewer: This should not be a `bytes1` but a `bool` here no?

```suggestion
  "valueType": "(address, bool)",
```

Reviewer: Also the question is how do we define an optional parameter in a schema? 🤔

>   "valueContent": "(Address, bool)"

Reviewer: This is not a valid value content

```suggestion
  "valueContent": "(Address, Boolean)"
```

![image](https://github.com/lukso-network/LIPs/assets/31145285/d571b28a-36c3-475b-9572-b14ed4e8c043)

0xcee78b4094da860110960000aabbccdd00000000000000000000000000000000
```

The boolean at the end can be used to specify whether the value sent to the extendable contract should be sent to the extension or stay in the extendable contract.

> For example, having the `0x01` at the end, signal that the boolean is true, which means that the value should be sent along the extension call.

Reviewer: ```suggestion
For example, having this boolean set to `true` (as a `0x01` hex byte) means that the value received should be forwarded to the extension.
```

```

> 0xcee78b4094da860110960000aabbccdd0000000000000000000000000000000001

Reviewer: This is incorrect. The bool byte should be appended to the data value, not the data key.

So for instance for the address `0xA8C62111e4652b07110A0FC81816303c42632f64` it would be constructed as follow:

```
                        address 
  vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
0xcafecafecafecafecafecafecafecafecafecafe01
                                          ^^
                                         true
```

```
                        address 
  vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
0xA8C62111e4652b07110A0FC81816303c42632f6401
                                          ^^
                                         true
```
